tests/fuzz_test.py

```diff
@@ -0,0 +1,133 @@
+
+
+class DSInputs:
+    """Fuzzer for HIDs attached to the driver station."""
+
+    def __init__(self) -> None:
+        self.gamepad = wpilib.simulation.XboxControllerSim(0)
+        self.joystick = wpilib.simulation.JoystickSim(1)
+
+    def fuzz(self) -> None:
+        fuzz_xbox_gamepad(self.gamepad)
+        fuzz_joystick(self.joystick)
+
+
+def fuzz_joystick(joystick: wpilib.simulation.JoystickSim) -> None:
+    """Fuzz a Logitech Extreme 3D Pro flight stick."""
+    for axis in range(5):
+        joystick.setRawAxis(axis, rand_axis())
+    for button in range(12):
+        joystick.setRawButton(button, rand_bool())
+    joystick.setPOV(rand_pov())
+
+
+def fuzz_xbox_gamepad(gamepad: wpilib.simulation.XboxControllerSim) -> None:
+    """Fuzz an XInput gamepad."""
+    gamepad.setLeftX(rand_axis())
+    gamepad.setLeftY(rand_axis())
+    gamepad.setRightX(rand_axis())
+    gamepad.setRightY(rand_axis())
+    gamepad.setLeftTriggerAxis(random.random())
+    gamepad.setRightTriggerAxis(random.random())
+    for button in range(10):
+        gamepad.setRawButton(button, rand_bool())
+    gamepad.setPOV(rand_pov())
+
+
+def _test_fuzz(control: TestController, fuzz_disabled_hids: bool) -> None:
+    with control.run_robot():
+        things = AllTheThings()
+        hids = DSInputs()
+
+        # Disabled mode
+        control.step_timing(seconds=0.2, autonomous=False, enabled=False)
+        things.fuzz()
+        if fuzz_disabled_hids:
+            hids.fuzz()
+        control.step_timing(seconds=0.2, autonomous=False, enabled=False)
+
+        # Autonomous mode
+        things.fuzz()
+        control.step_timing(seconds=0.2, autonomous=True, enabled=False)
+        things.fuzz()
+        control.step_timing(seconds=0.2, autonomous=True, enabled=True)
+
+        # Transition between autonomous and teleop
+        things.fuzz()
+        control.step_timing(seconds=0.2, autonomous=False, enabled=False)
+        things.fuzz()
+        control.step_timing(seconds=0.2, autonomous=False, enabled=True)
+
+        # Teleop
+        for _ in range(20):
+            things.fuzz()
+            hids.fuzz()
+            control.step_timing(seconds=0.1, autonomous=False, enabled=True)
+
+
+def test_fuzz(control: TestController) -> None:
+    _test_fuzz(control, fuzz_disabled_hids=False)
+
+
+def test_fuzz_disabled(control: TestController) -> None:
+    _test_fuzz(control, fuzz_disabled_hids=True)
+
+
+def test_fuzz_test(control: TestController) -> None:
+    from wpilib.simulation import DriverStationSim
+
+    with control.run_robot():
+        hids = DSInputs()
+
+        DriverStationSim.setDsAttached(True)
+        DriverStationSim.setAutonomous(False)
+        DriverStationSim.setTest(True)
+        DriverStationSim.setEnabled(True)
+
+        assert control.robot_is_alive
+
+        for _ in range(40):
+            hids.fuzz()
+            DriverStationSim.notifyNewData()
+            wpilib.simulation.stepTiming(0.05)
```

Comment:
I've increased the time step from `0.02` to `0.05`, to avoid this test deadlocking.